feat: add C implementation for `stats/base/dists/erlang/mode`

vivekmaurya001 opened the pull request:
Resolves #3574.

## Description

- adds C implementation for @stdlib/stats/base/dists/erlang/mode along with relevant tests, benchmarks and examples

This pull request:

- Progresses: https://github.com/stdlib-js/stdlib/issues/3574
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
feat(add c implementation): add stats/base/dists/erlang/mode

vivekmaurya001 committed:
feat(add c implementation): add stats/base/dists/erlang/mode

Planeshifter reviewed (changes requested):
This PR will have to be refactored to change `k` to be `int32_t` typed in the C implementation. See some of the other merged `erlang` distribution packages for reference: https://github.com/stdlib-js/stdlib/pulls?q=is%3Apr+erlang+is%3Aclosed

Would be great if you were able to take this on, @vivekmaurya001!

vivekmaurya001 committed:
Merge branch 'develop' into erlang/mode

Signed-off-by: Vivek maurya <155618190+vivekmaurya001@users.noreply.github.com>

vivekmaurya001 committed:
feat(add c implementation): add stats/base/dists/erlang/mode

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

vivekmaurya001 committed:
feat(add c implementation): add stats/base/dists/erlang/mode

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

vivekmaurya001 committed:
feat(add c implementation): add stats/base/dists/erlang/mode

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

vivekmaurya001 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/erlang/mode/README.md:
```suggestion
Returns the [mode][mode] of an [Erlang][erlang-distribution] distribution with parameters `k` (shape parameter) and `lambda` (rate parameter).
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/erlang/mode/src/main.c:
```suggestion
* double y = stdlib_base_dists_erlang_mode( 1, 1.0 );
```

Planeshifter committed:
chore: apply suggestions from code review

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Thanks, @vivekmaurya001!
